src/api/donations.py

```
@@ -0,0 +1,618 @@
+        if resp.ok:
+            print(f"[email] Thank-you sent to {to_email}")
+        else:
+            text = await resp.text()
+            print(f"[email] Mailgun error {resp.status}: {text}")
+    except Exception as exc:
+        print(f"[email] _send_thank_you_email error: {exc}")
+
+
+# DB helper shared by webhook handlers
+
+async def _mark_donation(env, field: str, value: str, status: str):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_mark_donation`

(ANN202)